### PR TITLE
modeler user + join character

### PR DESCRIPTION
**Pull request by Erikad94**
issue #25

### CODE REVIEW

**Comment by Baltazar510** [comment]
:+1:

**Comment by pellerinse2302** [comment]
:+1:

File: backend/app/models/user.js [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+UserSchema.pre('save', function(next){
+	var user = this;
+
+    // only hash the password if it has been modified (or is new)
+	//console.log(user.isModified('hashPassword'));
+    //if (!user.isModified('hashPassword')) return next();
+
+    // generate a salt
+    bcrypt.genSalt(SALT_WORK_FACTOR, function(err, salt) {
```
Comment (MicLamb) [inline backend/app/models/user.js]: Is the salt saved in the database to compare the hashed passwords? I think it should be saved so we could test if it is working! :smile: 

The rest looks good to meh.

Comment (DrPandemic) [inline backend/app/models/user.js]: Bcrypt manage all this magic! The salt is part of the hash. This is an example : 
```
$2a$10$N9qo8uLOickgx2ZMRZoMyeIjZAgcfl7p92ldGxad68LJZdL17lhWy
```
the ```N9qo8uLOic``` is the salt

**Comment by DrPandemic** [comment]
I just failed... I wasn't planning to merge right now. Sorry I cant auto rollback this merge... My bad :-1:

**Comment by Erikad94** [comment]
hahahaha

**Pull request merged**